docs/explanations/default-options.rst

```diff
@@ -0,0 +1,63 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _explanations-default-options:
+
+*********************
+Default build options
+*********************
+
+Meson offers many `built-in options <https://mesonbuild.com/Builtin-options.html>`__,
+and in the vast majority of cases those have good defaults. There are a couple
+of cases however where ``meson-python`` either is forced or chooses to override
+those with its own defaults. To view what those are for the version of
+``meson-python`` you have installed, look at the *User defined options* section
+of the output during the configure stage of the build (e.g., by running
+``python -m build --wheel``). This will look something like:
+
+.. code-block:: text
+
+    User defined options
+      Native files     : /home/username/code/project/.mesonpy-native-file.ini
+      debug            : false
+      optimization     : 2
+      prefix           : /home/username/mambaforge/envs/project-dev
+      python.platlibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
+      python.purelibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
+      b_ndebug         : if-release
+
+Let's go through each option and why they are used:
+
+- meson-python uses a native file, named ``.mesonpy-native-file.ini`` in order
+  to point Meson at the correct ``python`` interpreter to use (the same one for
+  which ``meson-python`` was installed). This is necessary, because Meson may
+  otherwise look for the first Python interpreter on the PATH (usually the same
+  one, but not always the case).
+- The ``prefix`` and ``platlibdir``/``purelibdir`` options also point Meson at
+  that same interpreter and the environment in which it is installed.
```

Comment:
These are going away, see #279.

Reply:
That's merged, but these options are still there. They're not specific to editable installs, so I'm not sure I understand what you meant here.

Reply:
I think Daniele meant #340. I'll review that tomorrow, but we can either wait until that gets a resolution, or merge this and have that PR update it.

Reply:
Yes, sorry, I linked to the wrong PR.

Reply:
Because of #341 and #344, we cannot get rid of the these options at the moment, so they should stay for now.

Reply:
The current text is good as is given the current state of the repo, I think we can resolve this comment.